bin/plugin/commands/release.js

```diff
@@ -0,0 +1,817 @@
+					'.' +
+					( releaseBranchParsedVersion.patch + 1 );
+			}
+
+			await askForConfirmation(
+				'The Version to release is ' +
+					formats.success( version ) +
+					'. Proceed?',
+				true,
+				abortMessage
+			);
+		}
+	);
+
+	return {
+		version,
+		versionLabel: version,
+		releaseBranch,
+	};
+}
+
+/**
+ * Bump the version in the different files (package.json, package-lock.json...)
+ * and commit the changes.
+ *
+ * @param {string} gitWorkingDirectoryPath Git Working Directory Path.
+ * @param {string} version                 Version to use.
+ * @param {string} changelog               Changelog to use.
+ * @param {string} abortMessage            Abort message.
+ * @return {string} hash of the version bump commit.
+ */
+async function runBumpPluginVersionUpdateChangelogAndCommitStep(
+	gitWorkingDirectoryPath,
+	version,
+	changelog,
+	abortMessage
+) {
+	let commitHash;
+	await runStep( 'Updating the plugin version', abortMessage, async () => {
+		const packageJsonPath = gitWorkingDirectoryPath + '/package.json';
+		const packageLockPath = gitWorkingDirectoryPath + '/package-lock.json';
+		const pluginFilePath =
+			gitWorkingDirectoryPath + '/' + config.pluginEntryPoint;
+		const packageJson = readJSONFile( packageJsonPath );
+		const packageLock = readJSONFile( packageLockPath );
+		const newPackageJson = {
+			...packageJson,
+			version,
+		};
+		fs.writeFileSync(
+			packageJsonPath,
+			JSON.stringify( newPackageJson, null, '\t' ) + '\n'
+		);
+		const newPackageLock = {
+			...packageLock,
+			version,
+		};
+		fs.writeFileSync(
+			packageLockPath,
+			JSON.stringify( newPackageLock, null, '\t' ) + '\n'
+		);
+		const content = fs.readFileSync( pluginFilePath, 'utf8' );
+		fs.writeFileSync(
+			pluginFilePath,
+			content.replace(
+				' * Version: ' + packageJson.version,
+				' * Version: ' + version
+			)
+		);
+
+		// Update the content of the readme.txt file
+		const readmePath = gitWorkingDirectoryPath + '/readme.txt';
+		const readmeFileContent = fs.readFileSync( readmePath, 'utf8' );
+		const newReadmeContent =
+			readmeFileContent.substr(
+				0,
+				readmeFileContent.indexOf( '== Changelog ==' )
+			) +
+			'== Changelog ==\n\n' +
+			`To read the changelog for ${ config.name } ${ version }, please navigate to the <a href="${ config.wpRepositoryReleasesUrl }v${ version }">release page</a>.` +
+			'\n';
+		fs.writeFileSync( readmePath, newReadmeContent );
+
+		// Update the content of the changelog.txt file
+		const stableVersion = version.match( /[0-9]+\.[0-9]+\.[0-9]+/ )[ 0 ];
+		const changelogPath = gitWorkingDirectoryPath + '/changelog.txt';
+		const changelogFileContent = fs.readFileSync( changelogPath, 'utf8' );
+		const versionHeader = '= ' + version + ' =\n\n';
+		const regexToSearch = /=\s([0-9]+\.[0-9]+\.[0-9]+)(-rc\.[0-9]+)?\s=\n\n/g;
+		let lastDifferentVersionMatch = regexToSearch.exec(
+			changelogFileContent
+		);
+		if ( lastDifferentVersionMatch[ 1 ] === stableVersion ) {
```

Comment:
I know it existed previously, but not sure how much we should trust this to be a match:

```
bin/plugin/commands/release.js:252:8 - error TS2531: Object is possibly 'null'.

252  	if ( lastDifferentVersionMatch[ 1 ] === stableVersion ) {
```

Reply:
This is where I was proposing to just skip this step for RC and only update the changelog for the stable release without any check. We should fix that separately from the refactor though.